src/main/java/org/commonjava/util/partyline/spi/JoinableFilesystem.java

```diff
@@ -30,4 +36,8 @@ JoinableFile getFile( File file, LocalLockOwner lockOwner, StreamCallbacks callb
             throws IOException, InterruptedException;
 
     LocalLockManager getLocalLockManager();
+
+    void updateDominantLocks( String path, LockLevel level )
+                    throws SystemException, NotSupportedException, HeuristicRollbackException, HeuristicMixedException,
```

Comment:
This is the same as above, replace the transaction exceptions with a PartylineException